<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>ChIP-Seq Workflow Template • SYSPipeRChipseq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><meta property="og:title" content="ChIP-Seq Workflow Template">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SYSPipeRChipseq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a></a>
</li>
<li>
  <a href="../articles/systemPipeChIPseq.html">
    <span class="fas fa fas fa-book"></span>
     
    Workflow
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/tgirke/systemPipeR">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<form class="navbar-form navbar-right hidden-xs hidden-sm" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>ChIP-Seq Workflow Template</h1>
                        <h4 class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 class="date">Last update: 14 February, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/SYSPipeRChipseq/blob/master/vignettes/systemPipeChIPseq.Rmd"><code>vignettes/systemPipeChIPseq.Rmd</code></a></small>
      <div class="hidden name"><code>systemPipeChIPseq.Rmd</code></div>

    </div>

    
    
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<!--
- Compile from command-line
Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::html_document'), clean=F); knitr::knit('systemPipeChIPseq.Rmd', tangle=TRUE)"; Rscript ../md2jekyll.R systemPipeChIPseq.knit.md 13; Rscript -e "rmarkdown::render('systemPipeChIPseq.Rmd', c('BiocStyle::pdf_document'))"
--><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>Users want to provide here background information about the design of their ChIP-Seq project.</p>
<div id="background-and-objectives" class="section level2">
<h2 class="hasAnchor">
<a href="#background-and-objectives" class="anchor"></a>Background and objectives</h2>
<p>This report describes the analysis of several ChIP-Seq experiments studying the DNA binding patterns of the transcriptions factors … from <em>organism</em> ….</p>
</div>
<div id="experimental-design" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-design" class="anchor"></a>Experimental design</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div id="workflow-environment" class="section level1">
<h1 class="hasAnchor">
<a href="#workflow-environment" class="anchor"></a>Workflow environment</h1>
<div id="generate-workflow-environment" class="section level2">
<h2 class="hasAnchor">
<a href="#generate-workflow-environment" class="anchor"></a>Generate workflow environment</h2>
<p>Load workflow environment with sample data into your current working directory. The sample data are described <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#load-sample-data-and-workflow-templates">here</a>.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeRdata)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeRdata/man/genWorkenvir.html">genWorkenvir</a></span>(<span class="dt">workflow =</span> <span class="st">"chipseq"</span>)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/getwd.html">setwd</a></span>(<span class="st">"chipseq"</span>)</a></code></pre></div>
<p>Alternatively, this can be done from the command-line as follows:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode sh"><code class="sourceCode bash"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="ex">Rscript</span> -e <span class="st">"systemPipeRdata::genWorkenvir(workflow='chipseq')"</span></a></code></pre></div>
<p>In the workflow environments generated by <code>genWorkenvir</code> all data inputs are stored in a <code>data/</code> directory and all analysis results will be written to a separate <code>results/</code> directory, while the <code>systemPipeChIPseq.Rmd</code> script and the <code>targets</code> file are expected to be located in the parent directory. The R session is expected to run from this parent directory. Additional parameter files are stored under <code>param/</code>.</p>
<p>To work with real data, users want to organize their own data similarly and substitute all test data for their own data. To rerun an established workflow on new data, the initial <code>targets</code> file along with the corresponding FASTQ files are usually the only inputs the user needs to provide.</p>
</div>
<div id="run-workflow" class="section level2">
<h2 class="hasAnchor">
<a href="#run-workflow" class="anchor"></a>Run workflow</h2>
<p>Now open the R markdown script <code>systemPipeChIPseq.Rmd</code> in your R IDE (<em>e.g.</em> vim-r or RStudio) and run the workflow as outlined below.</p>
<div id="run-r-session-on-computer-node" class="section level3">
<h3 class="hasAnchor">
<a href="#run-r-session-on-computer-node" class="anchor"></a>Run R session on computer node</h3>
<p>After opening the <code>Rmd</code> file of this workflow in Vim and attaching a connected R session via the <code>F2</code> (or other) key, use the following command sequence to run your R session on a computer node.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/quit.html">q</a></span>(<span class="st">"no"</span>)  <span class="co"># closes R session on head node</span></a></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="sourceCode bash"><code class="sourceCode bash"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="ex">srun</span> --x11 --partition=short --mem=2gb --cpus-per-task 4 --ntasks 1 --time 2:00:00 --pty bash -l</a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="ex">module</span> load R/3.6.0</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="ex">R</span></a></code></pre></div>
<p>Now check whether your R session is running on a computer node of the cluster and assess your environment.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/system.html">system</a></span>(<span class="st">"hostname"</span>)  <span class="co"># should return name of a compute node starting with i or c </span></a>
<a class="sourceLine" id="cb5-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/getwd.html">getwd</a></span>()  <span class="co"># checks current working directory of R session</span></a>
<a class="sourceLine" id="cb5-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/list.files.html">dir</a></span>()  <span class="co"># returns content of current working directory</span></a></code></pre></div>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeR)</a></code></pre></div>
<pre><code>## Loading required package: Rsamtools</code></pre>
<pre><code>## Loading required package: GenomeInfoDb</code></pre>
<pre><code>## Loading required package: BiocGenerics</code></pre>
<pre><code>## Loading required package: parallel</code></pre>
<pre><code>## 
## Attaching package: 'BiocGenerics'</code></pre>
<pre><code>## The following objects are masked from 'package:parallel':
## 
##     clusterApply, clusterApplyLB, clusterCall,
##     clusterEvalQ, clusterExport, clusterMap,
##     parApply, parCapply, parLapply, parLapplyLB,
##     parRapply, parSapply, parSapplyLB</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     IQR, mad, sd, var, xtabs</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     anyDuplicated, append, as.data.frame, basename,
##     cbind, colnames, dirname, do.call, duplicated,
##     eval, evalq, Filter, Find, get, grep, grepl,
##     intersect, is.unsorted, lapply, Map, mapply,
##     match, mget, order, paste, pmax, pmax.int, pmin,
##     pmin.int, Position, rank, rbind, Reduce,
##     rownames, sapply, setdiff, sort, table, tapply,
##     union, unique, unsplit, which, which.max,
##     which.min</code></pre>
<pre><code>## Loading required package: S4Vectors</code></pre>
<pre><code>## Loading required package: stats4</code></pre>
<pre><code>## 
## Attaching package: 'S4Vectors'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     expand.grid</code></pre>
<pre><code>## Loading required package: IRanges</code></pre>
<pre><code>## Loading required package: GenomicRanges</code></pre>
<pre><code>## Loading required package: Biostrings</code></pre>
<pre><code>## Loading required package: XVector</code></pre>
<pre><code>## 
## Attaching package: 'Biostrings'</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     strsplit</code></pre>
<pre><code>## Loading required package: ShortRead</code></pre>
<pre><code>## Loading required package: BiocParallel</code></pre>
<pre><code>## Loading required package: GenomicAlignments</code></pre>
<pre><code>## Loading required package: SummarizedExperiment</code></pre>
<pre><code>## Loading required package: Biobase</code></pre>
<pre><code>## Welcome to Bioconductor
## 
##     Vignettes contain introductory material; view
##     with 'browseVignettes()'. To cite Bioconductor,
##     see 'citation("Biobase")', and for packages
##     'citation("pkgname")'.</code></pre>
<pre><code>## Loading required package: DelayedArray</code></pre>
<pre><code>## Loading required package: matrixStats</code></pre>
<pre><code>## 
## Attaching package: 'matrixStats'</code></pre>
<pre><code>## The following objects are masked from 'package:Biobase':
## 
##     anyMissing, rowMedians</code></pre>
<pre><code>## 
## Attaching package: 'DelayedArray'</code></pre>
<pre><code>## The following objects are masked from 'package:matrixStats':
## 
##     colMaxs, colMins, colRanges, rowMaxs, rowMins,
##     rowRanges</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     aperm, apply, rowsum</code></pre>
<pre><code>## </code></pre>
<pre><code>## 
## Attaching package: 'systemPipeR'</code></pre>
<pre><code>## The following object is masked from 'package:BiocStyle':
## 
##     output</code></pre>
<p>If applicable users can load custom functions not provided by <code>systemPipeR</code>. Skip this step if this is not the case.</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb41-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/source.html">source</a></span>(<span class="st">"systemPipeChIPseq_Fct.R"</span>)</a></code></pre></div>
</div>
</div>
</div>
<div id="read-preprocessing" class="section level1">
<h1 class="hasAnchor">
<a href="#read-preprocessing" class="anchor"></a>Read preprocessing</h1>
<div id="experiment-definition-provided-by-targets-file" class="section level2">
<h2 class="hasAnchor">
<a href="#experiment-definition-provided-by-targets-file" class="anchor"></a>Experiment definition provided by <code>targets</code> file</h2>
<p>The <code>targets</code> file defines all FASTQ files and sample comparisons of the analysis workflow.</p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb42-1" data-line-number="1">targetspath &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets_chip.txt"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb42-2" data-line-number="2">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(targetspath, <span class="dt">comment.char =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb42-3" data-line-number="3">targets[<span class="dv">1</span><span class="op">:</span><span class="dv">4</span>, <span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">5</span>, <span class="dv">6</span>)]</a></code></pre></div>
<pre><code>##                      FileName SampleName Factor SampleLong
## 1 ./data/SRR446027_1.fastq.gz        M1A     M1  Mock.1h.A
## 2 ./data/SRR446028_1.fastq.gz        M1B     M1  Mock.1h.B
## 3 ./data/SRR446029_1.fastq.gz        A1A     A1   Avr.1h.A
## 4 ./data/SRR446030_1.fastq.gz        A1B     A1   Avr.1h.B
##   SampleReference
## 1                
## 2                
## 3             M1A
## 4             M1B</code></pre>
</div>
<div id="read-quality-filtering-and-trimming" class="section level2">
<h2 class="hasAnchor">
<a href="#read-quality-filtering-and-trimming" class="anchor"></a>Read quality filtering and trimming</h2>
<p>The following example shows how one can design a custom read preprocessing function using utilities provided by the <code>ShortRead</code> package, and then apply it with <code>preprocessReads</code> in batch mode to all FASTQ samples referenced in the corresponding <code>SYSargs2</code> instance (<code>trim</code> object below). More detailed information on read preprocessing is provided in <code>systemPipeR's</code> main vignette.</p>
<p>First, we construct <em><code>SYSargs2</code></em> object from <em><code>cwl</code></em> and <em><code>yml</code></em> param and <em><code>targets</code></em> files.</p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb44-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/preprocessReads/trim-se"</span>, </a>
<a class="sourceLine" id="cb44-2" data-line-number="2">    <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb44-3" data-line-number="3">trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> targetspath, <span class="dt">wf_file =</span> <span class="st">"trim-se.cwl"</span>, </a>
<a class="sourceLine" id="cb44-4" data-line-number="4">    <span class="dt">input_file =</span> <span class="st">"trim-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb44-5" data-line-number="5">trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(trim, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb44-6" data-line-number="6">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb44-7" data-line-number="7">trim</a>
<a class="sourceLine" id="cb44-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">output</a></span>(trim)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<p>Next, we execute the code for trimming all the raw data.</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb45-1" data-line-number="1"><span class="co"># args &lt;- systemArgs(sysma='param/trim.param',</span></a>
<a class="sourceLine" id="cb45-2" data-line-number="2"><span class="co"># mytargets='targets_chip.txt')</span></a>
<a class="sourceLine" id="cb45-3" data-line-number="3">filterFct &lt;-<span class="st"> </span><span class="cf">function</span>(fq, <span class="dt">cutoff =</span> <span class="dv">20</span>, <span class="dt">Nexceptions =</span> <span class="dv">0</span>) {</a>
<a class="sourceLine" id="cb45-4" data-line-number="4">    qcount &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span>(<span class="kw">as</span>(<span class="kw">quality</span>(fq), <span class="st">"matrix"</span>) <span class="op">&lt;=</span><span class="st"> </span>cutoff, <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb45-5" data-line-number="5">    fq[qcount <span class="op">&lt;=</span><span class="st"> </span>Nexceptions]</a>
<a class="sourceLine" id="cb45-6" data-line-number="6">    <span class="co"># Retains reads where Phred scores are &gt;= cutoff with N</span></a>
<a class="sourceLine" id="cb45-7" data-line-number="7">    <span class="co"># exceptions</span></a>
<a class="sourceLine" id="cb45-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb45-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/preprocessReads.html">preprocessReads</a></span>(<span class="dt">args =</span> trim, <span class="dt">Fct =</span> <span class="st">"filterFct(fq, cutoff=20, Nexceptions=0)"</span>, </a>
<a class="sourceLine" id="cb45-10" data-line-number="10">    <span class="dt">batchsize =</span> <span class="fl">1e+05</span>)</a>
<a class="sourceLine" id="cb45-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> trim, <span class="dt">file =</span> <span class="st">"targets_chip_trim.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb45-12" data-line-number="12">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="fastq-quality-report" class="section level2">
<h2 class="hasAnchor">
<a href="#fastq-quality-report" class="anchor"></a>FASTQ quality report</h2>
<p>The following <code>seeFastq</code> and <code>seeFastqPlot</code> functions generate and plot a series of useful quality statistics for a set of FASTQ files including per cycle quality box plots, base proportions, base-level quality trends, relative k-mer diversity, length and occurrence distribution of reads, number of reads above quality cutoffs and mean quality distribution. The results are written to a PDF file named <code>fastqReport.pdf</code>. Parallelization of FASTQ quality report via scheduler (<em>e.g.</em> Slurm) across several compute nodes.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb46-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BiocParallel)</a>
<a class="sourceLine" id="cb46-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(batchtools)</a>
<a class="sourceLine" id="cb46-3" data-line-number="3">f &lt;-<span class="st"> </span><span class="cf">function</span>(x) {</a>
<a class="sourceLine" id="cb46-4" data-line-number="4">    <span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(systemPipeR)</a>
<a class="sourceLine" id="cb46-5" data-line-number="5">    targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets_chip.txt"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb46-6" data-line-number="6">    dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/preprocessReads/trim-se"</span>, </a>
<a class="sourceLine" id="cb46-7" data-line-number="7">        <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb46-8" data-line-number="8">    trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="dt">targets =</span> targets, <span class="dt">wf_file =</span> <span class="st">"trim-se.cwl"</span>, </a>
<a class="sourceLine" id="cb46-9" data-line-number="9">        <span class="dt">input_file =</span> <span class="st">"trim-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb46-10" data-line-number="10">    trim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(trim, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb46-11" data-line-number="11">        <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb46-12" data-line-number="12">    <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastq</a></span>(<span class="dt">fastq =</span> <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(trim)[x], <span class="dt">batchsize =</span> <span class="fl">1e+05</span>, <span class="dt">klength =</span> <span class="dv">8</span>)</a>
<a class="sourceLine" id="cb46-13" data-line-number="13">}</a>
<a class="sourceLine" id="cb46-14" data-line-number="14"></a>
<a class="sourceLine" id="cb46-15" data-line-number="15">resources &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">walltime =</span> <span class="dv">120</span>, <span class="dt">ntasks =</span> <span class="dv">1</span>, <span class="dt">ncpus =</span> <span class="dv">4</span>, <span class="dt">memory =</span> <span class="dv">1024</span>)</a>
<a class="sourceLine" id="cb46-16" data-line-number="16">param &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/BiocParallel/man/BatchtoolsParam-class.html">BatchtoolsParam</a></span>(<span class="dt">workers =</span> <span class="dv">4</span>, <span class="dt">cluster =</span> <span class="st">"slurm"</span>, <span class="dt">template =</span> <span class="st">"batchtools.slurm.tmpl"</span>, </a>
<a class="sourceLine" id="cb46-17" data-line-number="17">    <span class="dt">resources =</span> resources)</a>
<a class="sourceLine" id="cb46-18" data-line-number="18">fqlist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/BiocParallel/man/bplapply.html">bplapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> trim), f, <span class="dt">BPPARAM =</span> param)</a>
<a class="sourceLine" id="cb46-19" data-line-number="19"></a>
<a class="sourceLine" id="cb46-20" data-line-number="20"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"./results/fastqReport.pdf"</span>, <span class="dt">height =</span> <span class="dv">18</span>, <span class="dt">width =</span> <span class="dv">4</span> <span class="op">*</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(fqlist))</a>
<a class="sourceLine" id="cb46-21" data-line-number="21"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/seeFastq.html">seeFastqPlot</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(fqlist, <span class="dt">recursive =</span> <span class="ot">FALSE</span>))</a>
<a class="sourceLine" id="cb46-22" data-line-number="22"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/fastqReport.png"><div align="center">
Figure 1: FASTQ quality report for 18 samples
</div>
<p></p>
</div>
</div>
<div id="alignments" class="section level1">
<h1 class="hasAnchor">
<a href="#alignments" class="anchor"></a>Alignments</h1>
<div id="read-mapping-with-bowtie2" class="section level2">
<h2 class="hasAnchor">
<a href="#read-mapping-with-bowtie2" class="anchor"></a>Read mapping with <code>Bowtie2</code>
</h2>
<p>The NGS reads of this project will be aligned with <code>Bowtie2</code> against the reference genome sequence <span class="citation">(Langmead and Salzberg 2012)</span>. The parameter settings of the aligner are defined in the <code>bowtie2-index.cwl</code> and <code>bowtie2-index.yml</code> files. In ChIP-Seq experiments it is usually more appropriate to eliminate reads mapping to multiple locations. To achieve this, users want to remove the argument setting <code>-k 50 non-deterministic</code> in the configuration files.</p>
<p>Building the index:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb47-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/bowtie2/bowtie2-idx"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb47-2" data-line-number="2">idx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWorkflow</a></span>(<span class="dt">targets =</span> <span class="ot">NULL</span>, <span class="dt">wf_file =</span> <span class="st">"bowtie2-index.cwl"</span>, </a>
<a class="sourceLine" id="cb47-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"bowtie2-index.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb47-4" data-line-number="4">idx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(idx)</a>
<a class="sourceLine" id="cb47-5" data-line-number="5">idx</a>
<a class="sourceLine" id="cb47-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">cmdlist</a></span>(idx)</a>
<a class="sourceLine" id="cb47-7" data-line-number="7"></a>
<a class="sourceLine" id="cb47-8" data-line-number="8"><span class="co">## Run in single machine</span></a>
<a class="sourceLine" id="cb47-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(idx, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>)</a></code></pre></div>
<p>The following submits 18 alignment jobs via a scheduler to a computer cluster.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb48-1" data-line-number="1">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets_chip.txt"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb48-2" data-line-number="2">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/bowtie2/bowtie2-se"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb48-3" data-line-number="3">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> targets, <span class="dt">wf_file =</span> <span class="st">"bowtie2-mapping-se.cwl"</span>, </a>
<a class="sourceLine" id="cb48-4" data-line-number="4">    <span class="dt">input_file =</span> <span class="st">"bowtie2-mapping-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb48-5" data-line-number="5">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb48-6" data-line-number="6">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb48-7" data-line-number="7">args</a>
<a class="sourceLine" id="cb48-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">cmdlist</a></span>(args)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a>
<a class="sourceLine" id="cb48-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">output</a></span>(args)[<span class="dv">1</span><span class="op">:</span><span class="dv">2</span>]</a></code></pre></div>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb49-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html">moduleload</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">modules</a></span>(args))  <span class="co"># Skip if a module system is not used</span></a>
<a class="sourceLine" id="cb49-2" data-line-number="2">resources &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">walltime =</span> <span class="dv">120</span>, <span class="dt">ntasks =</span> <span class="dv">1</span>, <span class="dt">ncpus =</span> <span class="dv">4</span>, <span class="dt">memory =</span> <span class="dv">1024</span>)</a>
<a class="sourceLine" id="cb49-3" data-line-number="3">reg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/clusterRun.html">clusterRun</a></span>(args, <span class="dt">FUN =</span> runCommandline, <span class="dt">more.args =</span> <span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="dt">args =</span> args, </a>
<a class="sourceLine" id="cb49-4" data-line-number="4">    <span class="dt">dir =</span> <span class="ot">FALSE</span>), <span class="dt">conffile =</span> <span class="st">".batchtools.conf.R"</span>, <span class="dt">template =</span> <span class="st">"batchtools.slurm.tmpl"</span>, </a>
<a class="sourceLine" id="cb49-5" data-line-number="5">    <span class="dt">Njobs =</span> <span class="dv">18</span>, <span class="dt">runid =</span> <span class="st">"01"</span>, <span class="dt">resourceList =</span> resources)</a>
<a class="sourceLine" id="cb49-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/pkg/batchtools/man/getStatus.html">getStatus</a></span>(<span class="dt">reg =</span> reg)</a>
<a class="sourceLine" id="cb49-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/batchtools/man/waitForJobs.html">waitForJobs</a></span>(<span class="dt">reg =</span> reg)</a>
<a class="sourceLine" id="cb49-8" data-line-number="8">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/output_update.html">output_update</a></span>(args, <span class="dt">dir =</span> <span class="ot">FALSE</span>, <span class="dt">replace =</span> <span class="ot">TRUE</span>, <span class="dt">extension =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">".sam"</span>, </a>
<a class="sourceLine" id="cb49-9" data-line-number="9">    <span class="st">".bam"</span>))  <span class="co">## Updates the output(args) to the right location in the subfolders</span></a>
<a class="sourceLine" id="cb49-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">output</a></span>(args)</a></code></pre></div>
<p>Alternatively, one can run the alignments sequentially on a single system.</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb50-1" data-line-number="1">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">force =</span> F)</a></code></pre></div>
<p>Check whether all BAM files have been created and write out the new targets file.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb51-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_bam.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb51-2" data-line-number="2">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb51-3" data-line-number="3">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb51-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(outpaths)</a></code></pre></div>
</div>
<div id="read-and-alignment-stats" class="section level2">
<h2 class="hasAnchor">
<a href="#read-and-alignment-stats" class="anchor"></a>Read and alignment stats</h2>
<p>The following provides an overview of the number of reads in each sample and how many of them aligned to the reference.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb52-1" data-line-number="1">read_statsDF &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/alignStats.html">alignStats</a></span>(<span class="dt">args =</span> args)</a>
<a class="sourceLine" id="cb52-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(read_statsDF, <span class="st">"results/alignStats.xls"</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb52-3" data-line-number="3">    <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a>
<a class="sourceLine" id="cb52-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="st">"results/alignStats.xls"</span>)</a></code></pre></div>
</div>
<div id="create-symbolic-links-for-viewing-bam-files-in-igv" class="section level2">
<h2 class="hasAnchor">
<a href="#create-symbolic-links-for-viewing-bam-files-in-igv" class="anchor"></a>Create symbolic links for viewing BAM files in IGV</h2>
<p>The <code>symLink2bam</code> function creates symbolic links to view the BAM alignment files in a genome browser such as IGV without moving these large files to a local system. The corresponding URLs are written to a file with a path specified under <code>urlfile</code>, here <code>IGVurl.txt</code>. Please replace the directory and the user name.</p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb53-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/symLink2bam.html">symLink2bam</a></span>(<span class="dt">sysargs =</span> args, <span class="dt">htmldir =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"~/.html/"</span>, <span class="st">"somedir/"</span>), </a>
<a class="sourceLine" id="cb53-2" data-line-number="2">    <span class="dt">urlbase =</span> <span class="st">"http://cluster.hpcc.ucr.edu/~tgirke/"</span>, <span class="dt">urlfile =</span> <span class="st">"./results/IGVurl.txt"</span>)</a></code></pre></div>
</div>
</div>
<div id="utilities-for-coverage-data" class="section level1">
<h1 class="hasAnchor">
<a href="#utilities-for-coverage-data" class="anchor"></a>Utilities for coverage data</h1>
<p>The following introduces several utilities useful for ChIP-Seq data. They are not part of the actual workflow.</p>
<div id="rle-object-stores-coverage-information" class="section level2">
<h2 class="hasAnchor">
<a href="#rle-object-stores-coverage-information" class="anchor"></a>Rle object stores coverage information</h2>
<div class="sourceCode" id="cb54"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb54-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(rtracklayer)</a>
<a class="sourceLine" id="cb54-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicRanges)</a>
<a class="sourceLine" id="cb54-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Rsamtools)</a>
<a class="sourceLine" id="cb54-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicAlignments)</a>
<a class="sourceLine" id="cb54-5" data-line-number="5">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb54-6" data-line-number="6">aligns &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicAlignments/man/readGAlignments.html">readGAlignments</a></span>(outpaths[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb54-7" data-line-number="7">cov &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicRanges/man/coverage-methods.html">coverage</a></span>(aligns)</a>
<a class="sourceLine" id="cb54-8" data-line-number="8">cov</a></code></pre></div>
</div>
<div id="resizing-aligned-reads" class="section level2">
<h2 class="hasAnchor">
<a href="#resizing-aligned-reads" class="anchor"></a>Resizing aligned reads</h2>
<div class="sourceCode" id="cb55"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb55-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">trim</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/GenomicRanges/man/intra-range-methods.html">resize</a></span>(<span class="kw">as</span>(aligns, <span class="st">"GRanges"</span>), <span class="dt">width =</span> <span class="dv">200</span>))</a></code></pre></div>
</div>
<div id="naive-peak-calling" class="section level2">
<h2 class="hasAnchor">
<a href="#naive-peak-calling" class="anchor"></a>Naive peak calling</h2>
<div class="sourceCode" id="cb56"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb56-1" data-line-number="1">islands &lt;-<span class="st"> </span><span class="kw">slice</span>(cov, <span class="dt">lower =</span> <span class="dv">15</span>)</a>
<a class="sourceLine" id="cb56-2" data-line-number="2">islands[[<span class="dv">1</span>]]</a></code></pre></div>
</div>
<div id="plot-coverage-for-defined-region" class="section level2">
<h2 class="hasAnchor">
<a href="#plot-coverage-for-defined-region" class="anchor"></a>Plot coverage for defined region</h2>
<div class="sourceCode" id="cb57"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb57-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ggbio)</a>
<a class="sourceLine" id="cb57-2" data-line-number="2">myloc &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Chr1"</span>, <span class="dv">1</span>, <span class="fl">1e+05</span>)</a>
<a class="sourceLine" id="cb57-3" data-line-number="3">ga &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicAlignments/man/readGAlignments.html">readGAlignments</a></span>(outpaths[<span class="dv">1</span>], <span class="dt">use.names =</span> <span class="ot">TRUE</span>, <span class="dt">param =</span> <span class="kw"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html">ScanBamParam</a></span>(<span class="dt">which =</span> <span class="kw"><a href="https://rdrr.io/pkg/GenomicRanges/man/GRanges-class.html">GRanges</a></span>(myloc[<span class="dv">1</span>], </a>
<a class="sourceLine" id="cb57-4" data-line-number="4">    <span class="kw">IRanges</span>(<span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(myloc[<span class="dv">2</span>]), <span class="kw"><a href="https://rdrr.io/r/base/numeric.html">as.numeric</a></span>(myloc[<span class="dv">3</span>])))))</a>
<a class="sourceLine" id="cb57-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/pkg/ggbio/man/autoplot-method.html">autoplot</a></span>(ga, <span class="kw">aes</span>(<span class="dt">color =</span> strand, <span class="dt">fill =</span> strand), <span class="dt">facets =</span> strand <span class="op">~</span><span class="st"> </span></a>
<a class="sourceLine" id="cb57-6" data-line-number="6"><span class="st">    </span>seqnames, <span class="dt">stat =</span> <span class="st">"coverage"</span>)</a></code></pre></div>
</div>
</div>
<div id="peak-calling-with-macs2" class="section level1">
<h1 class="hasAnchor">
<a href="#peak-calling-with-macs2" class="anchor"></a>Peak calling with MACS2</h1>
<div id="merge-bam-files-of-replicates-prior-to-peak-calling" class="section level2">
<h2 class="hasAnchor">
<a href="#merge-bam-files-of-replicates-prior-to-peak-calling" class="anchor"></a>Merge BAM files of replicates prior to peak calling</h2>
<p>Merging BAM files of technical and/or biological replicates can improve the sensitivity of the peak calling by increasing the depth of read coverage. The <code>mergeBamByFactor</code> function merges BAM files based on grouping information specified by a <code>factor</code>, here the <code>Factor</code> column of the imported targets file. It also returns an updated <code>SYSargs2</code> object containing the paths to the merged BAM files as well as to any unmerged files without replicates. This step can be skipped if merging of BAM files is not desired.</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb58-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/mergeBamByFactor"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb58-2" data-line-number="2">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_bam.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"merge-bam.cwl"</span>, </a>
<a class="sourceLine" id="cb58-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"merge-bam.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb58-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_BAM_PATH_"</span>, </a>
<a class="sourceLine" id="cb58-5" data-line-number="5">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb58-6" data-line-number="6"></a>
<a class="sourceLine" id="cb58-7" data-line-number="7">args_merge &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/mergeBamByFactor.html">mergeBamByFactor</a></span>(<span class="dt">args =</span> args, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb58-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args_merge, <span class="dt">file =</span> <span class="st">"targets_mergeBamByFactor.txt"</span>, </a>
<a class="sourceLine" id="cb58-9" data-line-number="9">    <span class="dt">step =</span> <span class="dv">1</span>, <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb58-10" data-line-number="10">    <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb59-1" data-line-number="1"><span class="co"># Skip if a module system is not used</span></a>
<a class="sourceLine" id="cb59-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/module.html">module</a></span>(<span class="st">"list"</span>)</a>
<a class="sourceLine" id="cb59-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/module.html">module</a></span>(<span class="st">"unload"</span>, <span class="st">"miniconda2"</span>)</a>
<a class="sourceLine" id="cb59-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/module.html">module</a></span>(<span class="st">"load"</span>, <span class="st">"python/2.7.14"</span>)  <span class="co"># Make sure to set up your enviroment variable for MACS2 </span></a></code></pre></div>
</div>
<div id="peak-calling-without-inputreference-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#peak-calling-without-inputreference-sample" class="anchor"></a>Peak calling without input/reference sample</h2>
<p>MACS2 can perform peak calling on ChIP-Seq data with and without input samples <span class="citation">(Zhang et al. 2008)</span>. The following performs peak calling without input on all samples specified in the corresponding <code>args</code> object. Note, due to the small size of the sample data, MACS2 needs to be run here with the <code>nomodel</code> setting. For real data sets, users want to remove this parameter in the corresponding <code>*.param</code> file(s).</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb60-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/MACS2/MACS2-noinput/"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb60-2" data-line-number="2">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_mergeBamByFactor.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"macs2.cwl"</span>, </a>
<a class="sourceLine" id="cb60-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"macs2.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb60-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb60-5" data-line-number="5">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb60-6" data-line-number="6"></a>
<a class="sourceLine" id="cb60-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>, <span class="dt">force =</span> T)</a>
<a class="sourceLine" id="cb60-8" data-line-number="8">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb60-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(outpaths)</a>
<a class="sourceLine" id="cb60-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_macs.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb60-11" data-line-number="11">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="peak-calling-with-inputreference-sample" class="section level2">
<h2 class="hasAnchor">
<a href="#peak-calling-with-inputreference-sample" class="anchor"></a>Peak calling with input/reference sample</h2>
<p>To perform peak calling with input samples, they can be most conveniently specified in the <code>SampleReference</code> column of the initial <code>targets</code> file. The <code>writeTargetsRef</code> function uses this information to create a <code>targets</code> file intermediate for running MACS2 with the corresponding input samples.</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb61-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsRef.html">writeTargetsRef</a></span>(<span class="dt">infile =</span> <span class="st">"targets_mergeBamByFactor.txt"</span>, <span class="dt">outfile =</span> <span class="st">"targets_bam_ref.txt"</span>, </a>
<a class="sourceLine" id="cb61-2" data-line-number="2">    <span class="dt">silent =</span> <span class="ot">FALSE</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb61-3" data-line-number="3">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/MACS2/MACS2-input/"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb61-4" data-line-number="4">args_input &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_bam_ref.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"macs2-input.cwl"</span>, </a>
<a class="sourceLine" id="cb61-5" data-line-number="5">    <span class="dt">input_file =</span> <span class="st">"macs2.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb61-6" data-line-number="6">args_input &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args_input, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName1 =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb61-7" data-line-number="7">    <span class="dt">FileName2 =</span> <span class="st">"_FASTQ_PATH2_"</span>, <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb61-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargs2list-methods.html">cmdlist</a></span>(args_input)[<span class="dv">1</span>]</a>
<a class="sourceLine" id="cb61-9" data-line-number="9"><span class="co">## Run</span></a>
<a class="sourceLine" id="cb61-10" data-line-number="10">args_input &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runCommandline.html">runCommandline</a></span>(args_input, <span class="dt">make_bam =</span> <span class="ot">FALSE</span>, <span class="dt">force =</span> T)</a>
<a class="sourceLine" id="cb61-11" data-line-number="11">outpaths_input &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args_input, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb61-12" data-line-number="12">    <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb61-13" data-line-number="13"><span class="kw"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span>(outpaths_input)</a>
<a class="sourceLine" id="cb61-14" data-line-number="14"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args_input, <span class="dt">file =</span> <span class="st">"targets_macs_input.txt"</span>, </a>
<a class="sourceLine" id="cb61-15" data-line-number="15">    <span class="dt">step =</span> <span class="dv">1</span>, <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb61-16" data-line-number="16">    <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>The peak calling results from MACS2 are written for each sample to separate files in the <code>results</code> directory. They are named after the corresponding files with extensions used by MACS2.</p>
</div>
<div id="identify-consensus-peaks" class="section level2">
<h2 class="hasAnchor">
<a href="#identify-consensus-peaks" class="anchor"></a>Identify consensus peaks</h2>
<p>The following example shows how one can identify consensus preaks among two peak sets sharing either a minimum absolute overlap and/or minimum relative overlap using the <code>subsetByOverlaps</code> or <code>olRanges</code> functions, respectively. Note, the latter is a custom function imported below by sourcing it.</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb62-1" data-line-number="1"><span class="co"># source('http://faculty.ucr.edu/~tgirke/Documents/R_BioCond/My_R_Scripts/rangeoverlapper.R')</span></a>
<a class="sourceLine" id="cb62-2" data-line-number="2">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)  <span class="co">## escolher um dos outputs index</span></a>
<a class="sourceLine" id="cb62-3" data-line-number="3">peak_M1A &lt;-<span class="st"> </span>outpaths[<span class="st">"M1A"</span>]</a>
<a class="sourceLine" id="cb62-4" data-line-number="4">peak_M1A &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(peak_M1A, <span class="dt">comment =</span> <span class="st">"#"</span>)[, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="st">"GRanges"</span>)</a>
<a class="sourceLine" id="cb62-5" data-line-number="5">peak_A1A &lt;-<span class="st"> </span>outpaths[<span class="st">"A1A"</span>]</a>
<a class="sourceLine" id="cb62-6" data-line-number="6">peak_A1A &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(peak_A1A, <span class="dt">comment =</span> <span class="st">"#"</span>)[, <span class="dv">1</span><span class="op">:</span><span class="dv">3</span>], <span class="st">"GRanges"</span>)</a>
<a class="sourceLine" id="cb62-7" data-line-number="7">(myol1 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicRanges/man/findOverlaps-methods.html">subsetByOverlaps</a></span>(peak_M1A, peak_A1A, <span class="dt">minoverlap =</span> <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb62-8" data-line-number="8"><span class="co"># Returns any overlap</span></a>
<a class="sourceLine" id="cb62-9" data-line-number="9">myol2 &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/olRanges.html">olRanges</a></span>(<span class="dt">query =</span> peak_M1A, <span class="dt">subject =</span> peak_A1A, <span class="dt">output =</span> <span class="st">"gr"</span>)</a>
<a class="sourceLine" id="cb62-10" data-line-number="10"><span class="co"># Returns any overlap with OL length information</span></a>
<a class="sourceLine" id="cb62-11" data-line-number="11">myol2[<span class="kw">values</span>(myol2)[<span class="st">"OLpercQ"</span>][, <span class="dv">1</span>] <span class="op">&gt;=</span><span class="st"> </span><span class="dv">50</span>]</a>
<a class="sourceLine" id="cb62-12" data-line-number="12"><span class="co"># Returns only query peaks with a minimum overlap of 50%</span></a></code></pre></div>
</div>
</div>
<div id="annotate-peaks-with-genomic-context" class="section level1">
<h1 class="hasAnchor">
<a href="#annotate-peaks-with-genomic-context" class="anchor"></a>Annotate peaks with genomic context</h1>
<div id="annotation-with-chippeakanno-package" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-with-chippeakanno-package" class="anchor"></a>Annotation with <code>ChIPpeakAnno</code> package</h2>
<p>The following annotates the identified peaks with genomic context information using the <code>ChIPpeakAnno</code> and <code>ChIPseeker</code> packages, respectively <span class="citation">(Zhu et al. 2010; Yu, Wang, and He 2015)</span>.</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb63-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ChIPpeakAnno)</a>
<a class="sourceLine" id="cb63-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicFeatures)</a>
<a class="sourceLine" id="cb63-3" data-line-number="3">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/annotate_peaks"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb63-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_macs.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"annotate-peaks.cwl"</span>, </a>
<a class="sourceLine" id="cb63-5" data-line-number="5">    <span class="dt">input_file =</span> <span class="st">"annotate-peaks.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb63-6" data-line-number="6">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb63-7" data-line-number="7">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb63-8" data-line-number="8"></a>
<a class="sourceLine" id="cb63-9" data-line-number="9">txdb &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/makeTxDbFromGFF.html">makeTxDbFromGFF</a></span>(<span class="dt">file =</span> <span class="st">"data/tair10.gff"</span>, <span class="dt">format =</span> <span class="st">"gff"</span>, </a>
<a class="sourceLine" id="cb63-10" data-line-number="10">    <span class="dt">dataSource =</span> <span class="st">"TAIR"</span>, <span class="dt">organism =</span> <span class="st">"Arabidopsis thaliana"</span>)</a>
<a class="sourceLine" id="cb63-11" data-line-number="11">ge &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span>(txdb, <span class="dt">columns =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"tx_name"</span>, <span class="st">"gene_id"</span>, <span class="st">"tx_type"</span>))</a>
<a class="sourceLine" id="cb63-12" data-line-number="12"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> args)) {</a>
<a class="sourceLine" id="cb63-13" data-line-number="13">    peaksGR &lt;-<span class="st"> </span><span class="kw">as</span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(args)[i], <span class="dt">comment =</span> <span class="st">"#"</span>), </a>
<a class="sourceLine" id="cb63-14" data-line-number="14">        <span class="st">"GRanges"</span>)</a>
<a class="sourceLine" id="cb63-15" data-line-number="15">    annotatedPeak &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/ChIPpeakAnno/man/annotatePeakInBatch.html">annotatePeakInBatch</a></span>(peaksGR, <span class="dt">AnnotationData =</span> <span class="kw"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span>(txdb))</a>
<a class="sourceLine" id="cb63-16" data-line-number="16">    df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(annotatedPeak), <span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw">values</span>(ge[<span class="kw">values</span>(annotatedPeak)<span class="op">$</span>feature, </a>
<a class="sourceLine" id="cb63-17" data-line-number="17">        ])))</a>
<a class="sourceLine" id="cb63-18" data-line-number="18">    outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb63-19" data-line-number="19">    <span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(df, outpaths[i], <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb63-20" data-line-number="20">        <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a>
<a class="sourceLine" id="cb63-21" data-line-number="21">}</a>
<a class="sourceLine" id="cb63-22" data-line-number="22"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_peakanno.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb63-23" data-line-number="23">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>The peak annotation results are written for each peak set to separate files in the <code>results</code> directory. They are named after the corresponding peak files with extensions specified in the <code>annotate_peaks.param</code> file, here <code>*.peaks.annotated.xls</code>.</p>
</div>
<div id="annotation-with-chipseeker-package" class="section level2">
<h2 class="hasAnchor">
<a href="#annotation-with-chipseeker-package" class="anchor"></a>Annotation with <code>ChIPseeker</code> package</h2>
<p>Same as in previous step but using the <code>ChIPseeker</code> package for annotating the peaks.</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb64-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(ChIPseeker)</a>
<a class="sourceLine" id="cb64-2" data-line-number="2"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> args)) {</a>
<a class="sourceLine" id="cb64-3" data-line-number="3">    peakAnno &lt;-<span class="st"> </span><span class="kw">annotatePeak</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(args)[i], <span class="dt">TxDb =</span> txdb, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb64-4" data-line-number="4">    df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(peakAnno)</a>
<a class="sourceLine" id="cb64-5" data-line-number="5">    outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb64-6" data-line-number="6">    <span class="kw"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span>(df, outpaths[i], <span class="dt">quote =</span> <span class="ot">FALSE</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb64-7" data-line-number="7">        <span class="dt">sep =</span> <span class="st">"</span><span class="ch">\t</span><span class="st">"</span>)</a>
<a class="sourceLine" id="cb64-8" data-line-number="8">}</a>
<a class="sourceLine" id="cb64-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_peakanno.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb64-10" data-line-number="10">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
<p>Summary plots provided by the <code>ChIPseeker</code> package. Here applied only to one sample for demonstration purposes.</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb65-1" data-line-number="1">peak &lt;-<span class="st"> </span><span class="kw">readPeakFile</span>(<span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(args)[<span class="dv">1</span>])</a>
<a class="sourceLine" id="cb65-2" data-line-number="2"><span class="kw">covplot</span>(peak, <span class="dt">weightCol =</span> <span class="st">"X.log10.pvalue."</span>)</a>
<a class="sourceLine" id="cb65-3" data-line-number="3">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb65-4" data-line-number="4"><span class="kw">peakHeatmap</span>(outpaths[<span class="dv">1</span>], <span class="dt">TxDb =</span> txdb, <span class="dt">upstream =</span> <span class="dv">1000</span>, <span class="dt">downstream =</span> <span class="dv">1000</span>, </a>
<a class="sourceLine" id="cb65-5" data-line-number="5">    <span class="dt">color =</span> <span class="st">"red"</span>)</a>
<a class="sourceLine" id="cb65-6" data-line-number="6"><span class="kw">plotAvgProf2</span>(outpaths[<span class="dv">1</span>], <span class="dt">TxDb =</span> txdb, <span class="dt">upstream =</span> <span class="dv">1000</span>, <span class="dt">downstream =</span> <span class="dv">1000</span>, </a>
<a class="sourceLine" id="cb65-7" data-line-number="7">    <span class="dt">xlab =</span> <span class="st">"Genomic Region (5'-&gt;3')"</span>, <span class="dt">ylab =</span> <span class="st">"Read Count Frequency"</span>)</a></code></pre></div>
</div>
</div>
<div id="count-reads-overlapping-peaks" class="section level1">
<h1 class="hasAnchor">
<a href="#count-reads-overlapping-peaks" class="anchor"></a>Count reads overlapping peaks</h1>
<p>The <code>countRangeset</code> function is a convenience wrapper to perform read counting iteratively over serveral range sets, here peak range sets. Internally, the read counting is performed with the <code>summarizeOverlaps</code> function from the <code>GenomicAlignments</code> package. The resulting count tables are directly saved to files, one for each peak set.</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb66-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(GenomicRanges)</a>
<a class="sourceLine" id="cb66-2" data-line-number="2">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/count_rangesets"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb66-3" data-line-number="3">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_macs.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"count_rangesets.cwl"</span>, </a>
<a class="sourceLine" id="cb66-4" data-line-number="4">    <span class="dt">input_file =</span> <span class="st">"count_rangesets.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb66-5" data-line-number="5">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb66-6" data-line-number="6">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb66-7" data-line-number="7"></a>
<a class="sourceLine" id="cb66-8" data-line-number="8"><span class="co">## Bam Files</span></a>
<a class="sourceLine" id="cb66-9" data-line-number="9">targets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata"</span>, <span class="st">"targets_chip.txt"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb66-10" data-line-number="10">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/bowtie2/bowtie2-se"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb66-11" data-line-number="11">args_bam &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> targets, <span class="dt">wf_file =</span> <span class="st">"bowtie2-mapping-se.cwl"</span>, </a>
<a class="sourceLine" id="cb66-12" data-line-number="12">    <span class="dt">input_file =</span> <span class="st">"bowtie2-mapping-se.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb66-13" data-line-number="13">args_bam &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args_bam, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb66-14" data-line-number="14">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb66-15" data-line-number="15">args_bam &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/output_update.html">output_update</a></span>(args_bam, <span class="dt">dir =</span> <span class="ot">FALSE</span>, <span class="dt">replace =</span> <span class="ot">TRUE</span>, </a>
<a class="sourceLine" id="cb66-16" data-line-number="16">    <span class="dt">extension =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">".sam"</span>, <span class="st">".bam"</span>))</a>
<a class="sourceLine" id="cb66-17" data-line-number="17">outpaths &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args_bam, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb66-18" data-line-number="18"></a>
<a class="sourceLine" id="cb66-19" data-line-number="19">bfl &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html">BamFileList</a></span>(outpaths, <span class="dt">yieldSize =</span> <span class="dv">50000</span>, <span class="dt">index =</span> <span class="kw"><a href="https://rdrr.io/r/base/character.html">character</a></span>())</a>
<a class="sourceLine" id="cb66-20" data-line-number="20">countDFnames &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/countRangeset.html">countRangeset</a></span>(bfl, args, <span class="dt">mode =</span> <span class="st">"Union"</span>, <span class="dt">ignore.strand =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb66-21" data-line-number="21"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args, <span class="dt">file =</span> <span class="st">"targets_countDF.txt"</span>, <span class="dt">step =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb66-22" data-line-number="22">    <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="differential-binding-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#differential-binding-analysis" class="anchor"></a>Differential binding analysis</h1>
<p>The <code>runDiff</code> function performs differential binding analysis in batch mode for several count tables using <code>edgeR</code> or <code>DESeq2</code> <span class="citation">(Robinson, McCarthy, and Smyth 2010; Love, Huber, and Anders 2014)</span>. Internally, it calls the functions <code>run_edgeR</code> and <code>run_DESeq2</code>. It also returns the filtering results and plots from the downstream <code>filterDEGs</code> function using the fold change and FDR cutoffs provided under the <code>dbrfilter</code> argument.</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb67-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/rundiff"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb67-2" data-line-number="2">args_diff &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_countDF.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"rundiff.cwl"</span>, </a>
<a class="sourceLine" id="cb67-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"rundiff.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb67-4" data-line-number="4">args_diff &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args_diff, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb67-5" data-line-number="5">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb67-6" data-line-number="6"></a>
<a class="sourceLine" id="cb67-7" data-line-number="7">cmp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/readComp.html">readComp</a></span>(<span class="dt">file =</span> args_bam, <span class="dt">format =</span> <span class="st">"matrix"</span>)</a>
<a class="sourceLine" id="cb67-8" data-line-number="8">dbrlist &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/runDiff.html">runDiff</a></span>(<span class="dt">args =</span> args_diff, <span class="dt">diffFct =</span> run_edgeR, <span class="dt">targets =</span> <span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/targets.as.df.html">targets.as.df</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/rtracklayer/man/targets.html">targets</a></span>(args_bam)), </a>
<a class="sourceLine" id="cb67-9" data-line-number="9">    <span class="dt">cmp =</span> cmp[[<span class="dv">1</span>]], <span class="dt">independent =</span> <span class="ot">TRUE</span>, <span class="dt">dbrfilter =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">Fold =</span> <span class="dv">2</span>, </a>
<a class="sourceLine" id="cb67-10" data-line-number="10">        <span class="dt">FDR =</span> <span class="dv">1</span>))</a>
<a class="sourceLine" id="cb67-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/writeTargetsout.html">writeTargetsout</a></span>(<span class="dt">x =</span> args_diff, <span class="dt">file =</span> <span class="st">"targets_rundiff.txt"</span>, </a>
<a class="sourceLine" id="cb67-12" data-line-number="12">    <span class="dt">step =</span> <span class="dv">1</span>, <span class="dt">new_col =</span> <span class="st">"FileName"</span>, <span class="dt">new_col_output_index =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb67-13" data-line-number="13">    <span class="dt">overwrite =</span> <span class="ot">TRUE</span>)</a></code></pre></div>
</div>
<div id="go-term-enrichment-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#go-term-enrichment-analysis" class="anchor"></a>GO term enrichment analysis</h1>
<p>The following performs GO term enrichment analysis for each annotated peak set.</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb68-1" data-line-number="1">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/annotate_peaks"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb68-2" data-line-number="2">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_bam_ref.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"annotate-peaks.cwl"</span>, </a>
<a class="sourceLine" id="cb68-3" data-line-number="3">    <span class="dt">input_file =</span> <span class="st">"annotate-peaks.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb68-4" data-line-number="4">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName1 =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb68-5" data-line-number="5">    <span class="dt">FileName2 =</span> <span class="st">"_FASTQ_PATH2_"</span>, <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb68-6" data-line-number="6"></a>
<a class="sourceLine" id="cb68-7" data-line-number="7">args_anno &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_macs.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"annotate-peaks.cwl"</span>, </a>
<a class="sourceLine" id="cb68-8" data-line-number="8">    <span class="dt">input_file =</span> <span class="st">"annotate-peaks.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb68-9" data-line-number="9">args_anno &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args_anno, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb68-10" data-line-number="10">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb68-11" data-line-number="11">annofiles &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/subsetWF.html">subsetWF</a></span>(args_anno, <span class="dt">slot =</span> <span class="st">"output"</span>, <span class="dt">subset =</span> <span class="dv">1</span>, </a>
<a class="sourceLine" id="cb68-12" data-line-number="12">    <span class="dt">index =</span> <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb68-13" data-line-number="13">gene_ids &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(annofiles), <span class="cf">function</span>(x) <span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(annofiles[x])[, </a>
<a class="sourceLine" id="cb68-14" data-line-number="14">    <span class="st">"geneId"</span>])), <span class="dt">simplify =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb68-15" data-line-number="15"><span class="kw"><a href="https://rdrr.io/r/base/load.html">load</a></span>(<span class="st">"data/GO/catdb.RData"</span>)</a>
<a class="sourceLine" id="cb68-16" data-line-number="16">BatchResult &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/GOHyperGAll.html">GOCluster_Report</a></span>(<span class="dt">catdb =</span> catdb, <span class="dt">setlist =</span> gene_ids, </a>
<a class="sourceLine" id="cb68-17" data-line-number="17">    <span class="dt">method =</span> <span class="st">"all"</span>, <span class="dt">id_type =</span> <span class="st">"gene"</span>, <span class="dt">CLSZ =</span> <span class="dv">2</span>, <span class="dt">cutoff =</span> <span class="fl">0.9</span>, </a>
<a class="sourceLine" id="cb68-18" data-line-number="18">    <span class="dt">gocats =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"MF"</span>, <span class="st">"BP"</span>, <span class="st">"CC"</span>), <span class="dt">recordSpecGO =</span> <span class="ot">NULL</span>)</a></code></pre></div>
</div>
<div id="motif-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#motif-analysis" class="anchor"></a>Motif analysis</h1>
<div id="parse-dna-sequences-of-peak-regions-from-genome" class="section level2">
<h2 class="hasAnchor">
<a href="#parse-dna-sequences-of-peak-regions-from-genome" class="anchor"></a>Parse DNA sequences of peak regions from genome</h2>
<p>Enrichment analysis of known DNA binding motifs or <em>de novo</em> discovery of novel motifs requires the DNA sequences of the identified peak regions. To parse the corresponding sequences from the reference genome, the <code>getSeq</code> function from the <code>Biostrings</code> package can be used. The following example parses the sequences for each peak set and saves the results to separate FASTA files, one for each peak set. In addition, the sequences in the FASTA files are ranked (sorted) by increasing p-values as expected by some motif discovery tools, such as <code>BCRANK</code>.</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb69-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(Biostrings)</a>
<a class="sourceLine" id="cb69-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(seqLogo)</a>
<a class="sourceLine" id="cb69-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(BCRANK)</a>
<a class="sourceLine" id="cb69-4" data-line-number="4">dir_path &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span>(<span class="st">"extdata/cwl/annotate_peaks"</span>, <span class="dt">package =</span> <span class="st">"systemPipeR"</span>)</a>
<a class="sourceLine" id="cb69-5" data-line-number="5">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/loadWorkflow.html">loadWF</a></span>(<span class="dt">targets =</span> <span class="st">"targets_macs.txt"</span>, <span class="dt">wf_file =</span> <span class="st">"annotate-peaks.cwl"</span>, </a>
<a class="sourceLine" id="cb69-6" data-line-number="6">    <span class="dt">input_file =</span> <span class="st">"annotate-peaks.yml"</span>, <span class="dt">dir_path =</span> dir_path)</a>
<a class="sourceLine" id="cb69-7" data-line-number="7">args &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/renderWF.html">renderWF</a></span>(args, <span class="dt">inputvars =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dt">FileName =</span> <span class="st">"_FASTQ_PATH1_"</span>, </a>
<a class="sourceLine" id="cb69-8" data-line-number="8">    <span class="dt">SampleName =</span> <span class="st">"_SampleName_"</span>))</a>
<a class="sourceLine" id="cb69-9" data-line-number="9"></a>
<a class="sourceLine" id="cb69-10" data-line-number="10">rangefiles &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/systemPipeR/man/sysargs.html">infile1</a></span>(args)</a>
<a class="sourceLine" id="cb69-11" data-line-number="11"><span class="cf">for</span> (i <span class="cf">in</span> <span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dt">along =</span> rangefiles)) {</a>
<a class="sourceLine" id="cb69-12" data-line-number="12">    df &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/utils/read.table.html">read.delim</a></span>(rangefiles[i], <span class="dt">comment =</span> <span class="st">"#"</span>)</a>
<a class="sourceLine" id="cb69-13" data-line-number="13">    peaks &lt;-<span class="st"> </span><span class="kw">as</span>(df, <span class="st">"GRanges"</span>)</a>
<a class="sourceLine" id="cb69-14" data-line-number="14">    <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(peaks) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/character.html">as.character</a></span>(<span class="kw">seqnames</span>(peaks)), <span class="st">"_"</span>, </a>
<a class="sourceLine" id="cb69-15" data-line-number="15">        <span class="kw"><a href="https://rdrr.io/r/stats/start.html">start</a></span>(peaks), <span class="st">"-"</span>, <span class="kw"><a href="https://rdrr.io/r/stats/start.html">end</a></span>(peaks))</a>
<a class="sourceLine" id="cb69-16" data-line-number="16">    peaks &lt;-<span class="st"> </span>peaks[<span class="kw"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="kw">values</span>(peaks)<span class="op">$</span>X.log10.pvalue., <span class="dt">decreasing =</span> <span class="ot">TRUE</span>)]</a>
<a class="sourceLine" id="cb69-17" data-line-number="17">    pseq &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/Biostrings/man/getSeq.html">getSeq</a></span>(<span class="kw"><a href="https://rdrr.io/pkg/Rsamtools/man/FaFile-class.html">FaFile</a></span>(<span class="st">"./data/tair10.fasta"</span>), peaks)</a>
<a class="sourceLine" id="cb69-18" data-line-number="18">    <span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(pseq) &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(peaks)</a>
<a class="sourceLine" id="cb69-19" data-line-number="19">    <span class="kw"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html">writeXStringSet</a></span>(pseq, <span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(rangefiles[i], <span class="st">".fasta"</span>))</a>
<a class="sourceLine" id="cb69-20" data-line-number="20">}</a></code></pre></div>
</div>
<div id="motif-discovery-with-bcrank" class="section level2">
<h2 class="hasAnchor">
<a href="#motif-discovery-with-bcrank" class="anchor"></a>Motif discovery with <code>BCRANK</code>
</h2>
<p>The Bioconductor package <code>BCRANK</code> is one of the many tools available for <em>de novo</em> discovery of DNA binding motifs in peak regions of ChIP-Seq experiments. The given example applies this method on the first peak sample set and plots the sequence logo of the highest ranking motif.</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb70-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">0</span>)</a>
<a class="sourceLine" id="cb70-2" data-line-number="2">BCRANKout &lt;-<span class="st"> </span><span class="kw">bcrank</span>(<span class="kw"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span>(rangefiles[<span class="dv">1</span>], <span class="st">".fasta"</span>), <span class="dt">restarts =</span> <span class="dv">25</span>, </a>
<a class="sourceLine" id="cb70-3" data-line-number="3">    <span class="dt">use.P1 =</span> <span class="ot">TRUE</span>, <span class="dt">use.P2 =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb70-4" data-line-number="4"><span class="kw">toptable</span>(BCRANKout)</a>
<a class="sourceLine" id="cb70-5" data-line-number="5">topMotif &lt;-<span class="st"> </span><span class="kw">toptable</span>(BCRANKout, <span class="dv">1</span>)</a>
<a class="sourceLine" id="cb70-6" data-line-number="6">weightMatrix &lt;-<span class="st"> </span><span class="kw">pwm</span>(topMotif, <span class="dt">normalize =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb70-7" data-line-number="7">weightMatrixNormalized &lt;-<span class="st"> </span><span class="kw">pwm</span>(topMotif, <span class="dt">normalize =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb70-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span>(<span class="st">"results/seqlogo.pdf"</span>)</a>
<a class="sourceLine" id="cb70-9" data-line-number="9"><span class="kw">seqLogo</span>(weightMatrixNormalized)</a>
<a class="sourceLine" id="cb70-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span>()</a></code></pre></div>
<img src="results/seqlogo.png"><div align="center">
Figure 2: One of the motifs identified by <code>BCRANK</code>
</div>
<p></p>
</div>
</div>
<div id="version-information" class="section level1">
<h1 class="hasAnchor">
<a href="#version-information" class="anchor"></a>Version Information</h1>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb71-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a></code></pre></div>
<pre><code>## R Under development (unstable) (2019-12-19 r77606)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Pop!_OS 19.04
## 
## Matrix products: default
## BLAS:   /usr/local/lib/R/lib/libRblas.so
## LAPACK: /usr/local/lib/R/lib/libRlapack.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats4    parallel  stats     graphics  grDevices
## [6] utils     datasets  methods   base     
## 
## other attached packages:
##  [1] systemPipeR_1.21.2          ShortRead_1.45.3           
##  [3] GenomicAlignments_1.23.1    SummarizedExperiment_1.17.1
##  [5] DelayedArray_0.13.3         matrixStats_0.55.0         
##  [7] Biobase_2.47.2              BiocParallel_1.21.2        
##  [9] Rsamtools_2.3.4             Biostrings_2.55.4          
## [11] XVector_0.27.0              GenomicRanges_1.39.2       
## [13] GenomeInfoDb_1.23.13        IRanges_2.21.3             
## [15] S4Vectors_0.25.12           BiocGenerics_0.33.0        
## [17] BiocStyle_2.15.6           
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_1.4-1         rjson_0.2.20            
##   [3] hwriter_1.3.2            rprojroot_1.3-2         
##   [5] fs_1.3.1                 rstudioapi_0.11         
##   [7] bit64_0.9-7              AnnotationDbi_1.49.1    
##   [9] codetools_0.2-16         splines_4.0.0           
##  [11] knitr_1.28               annotate_1.65.1         
##  [13] GO.db_3.10.0             dbplyr_1.4.2            
##  [15] png_0.1-7                pheatmap_1.0.12         
##  [17] graph_1.65.1             BiocManager_1.30.10     
##  [19] compiler_4.0.0           httr_1.4.1              
##  [21] GOstats_2.53.0           backports_1.1.5         
##  [23] assertthat_0.2.1         Matrix_1.2-18           
##  [25] lazyeval_0.2.2           limma_3.43.3            
##  [27] formatR_1.7              htmltools_0.4.0         
##  [29] prettyunits_1.1.1        tools_4.0.0             
##  [31] gtable_0.3.0             glue_1.3.1              
##  [33] GenomeInfoDbData_1.2.2   Category_2.53.1         
##  [35] dplyr_0.8.4              batchtools_0.9.12       
##  [37] rappdirs_0.3.1           Rcpp_1.0.3              
##  [39] pkgdown_1.4.1            vctrs_0.2.2             
##  [41] debugme_1.1.0            rtracklayer_1.47.0      
##  [43] xfun_0.12                stringr_1.4.0           
##  [45] lifecycle_0.1.0          XML_3.99-0.3            
##  [47] edgeR_3.29.0             zlibbioc_1.33.1         
##  [49] MASS_7.3-51.5            scales_1.1.0            
##  [51] BSgenome_1.55.3          VariantAnnotation_1.33.0
##  [53] hms_0.5.3                RBGL_1.63.1             
##  [55] RColorBrewer_1.1-2       yaml_2.2.1              
##  [57] curl_4.3                 memoise_1.1.0           
##  [59] ggplot2_3.2.1            biomaRt_2.43.3          
##  [61] latticeExtra_0.6-29      stringi_1.4.5           
##  [63] RSQLite_2.2.0            genefilter_1.69.0       
##  [65] desc_1.2.0               checkmate_2.0.0         
##  [67] GenomicFeatures_1.39.4   rlang_0.4.4             
##  [69] pkgconfig_2.0.3          bitops_1.0-6            
##  [71] evaluate_0.14            lattice_0.20-38         
##  [73] purrr_0.3.3              bit_1.1-15.2            
##  [75] tidyselect_1.0.0         GSEABase_1.49.0         
##  [77] AnnotationForge_1.29.2   magrittr_1.5            
##  [79] bookdown_0.17            R6_2.4.1                
##  [81] base64url_1.4            DBI_1.1.0               
##  [83] withr_2.1.2              pillar_1.4.3            
##  [85] survival_3.1-8           RCurl_1.98-1.1          
##  [87] tibble_2.1.3             crayon_1.3.4            
##  [89] BiocFileCache_1.11.4     rmarkdown_2.1           
##  [91] jpeg_0.1-8.1             progress_1.2.2          
##  [93] locfit_1.5-9.1           grid_4.0.0              
##  [95] data.table_1.12.8        blob_1.2.1              
##  [97] Rgraphviz_2.31.0         digest_0.6.24           
##  [99] xtable_1.8-4             brew_1.0-6              
## [101] openssl_1.4.1            munsell_0.5.0           
## [103] askpass_1.1</code></pre>
</div>
<div id="funding" class="section level1">
<h1 class="hasAnchor">
<a href="#funding" class="anchor"></a>Funding</h1>
<p>This project was supported by funds from the National Institutes of Health (NIH) and the National Science Foundation (NSF).</p>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-H_Backman2016-bt">
<p>H Backman, Tyler W, and Thomas Girke. 2016. “systemPipeR: NGS workflow and report generation environment.” <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0">https://doi.org/10.1186/s12859-016-1241-0</a>.</p>
</div>
<div id="ref-Langmead2012-bs">
<p>Langmead, Ben, and Steven L Salzberg. 2012. “Fast Gapped-Read Alignment with Bowtie 2.” <em>Nat. Methods</em> 9 (4). Nature Publishing Group: 357–59. <a href="https://doi.org/10.1038/nmeth.1923">https://doi.org/10.1038/nmeth.1923</a>.</p>
</div>
<div id="ref-Love2014-sh">
<p>Love, Michael, Wolfgang Huber, and Simon Anders. 2014. “Moderated Estimation of Fold Change and Dispersion for RNA-seq Data with DESeq2.” <em>Genome Biol.</em> 15 (12): 550. <a href="https://doi.org/10.1186/s13059-014-0550-8">https://doi.org/10.1186/s13059-014-0550-8</a>.</p>
</div>
<div id="ref-Robinson2010-uk">
<p>Robinson, M D, D J McCarthy, and G K Smyth. 2010. “EdgeR: A Bioconductor Package for Differential Expression Analysis of Digital Gene Expression Data.” <em>Bioinformatics</em> 26 (1): 139–40. <a href="https://doi.org/10.1093/bioinformatics/btp616">https://doi.org/10.1093/bioinformatics/btp616</a>.</p>
</div>
<div id="ref-Yu2015-xu">
<p>Yu, Guangchuang, Li-Gen Wang, and Qing-Yu He. 2015. “ChIPseeker: An R/Bioconductor Package for ChIP Peak Annotation, Comparison and Visualization.” <em>Bioinformatics</em> 31 (14): 2382–3. <a href="https://doi.org/10.1093/bioinformatics/btv145">https://doi.org/10.1093/bioinformatics/btv145</a>.</p>
</div>
<div id="ref-Zhang2008-pc">
<p>Zhang, Y, T Liu, C A Meyer, J Eeckhoute, D S Johnson, B E Bernstein, C Nussbaum, et al. 2008. “Model-Based Analysis of ChIP-Seq (MACS).” <em>Genome Biol.</em> 9 (9). <a href="https://doi.org/10.1186/gb-2008-9-9-r137">https://doi.org/10.1186/gb-2008-9-9-r137</a>.</p>
</div>
<div id="ref-Zhu2010-zo">
<p>Zhu, Lihua J, Claude Gazin, Nathan D Lawson, Hervé Pagès, Simon M Lin, David S Lapointe, and Michael R Green. 2010. “ChIPpeakAnno: A Bioconductor Package to Annotate ChIP-seq and ChIP-chip Data.” <em>BMC Bioinformatics</em> 11: 237. <a href="https://doi.org/10.1186/1471-2105-11-237">https://doi.org/10.1186/1471-2105-11-237</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#introduction">Introduction</a><ul class="nav nav-pills nav-stacked">
<li><a href="#background-and-objectives">Background and objectives</a></li>
      <li><a href="#experimental-design">Experimental design</a></li>
      </ul>
</li>
      <li>
<a href="#workflow-environment">Workflow environment</a><ul class="nav nav-pills nav-stacked">
<li><a href="#generate-workflow-environment">Generate workflow environment</a></li>
      <li><a href="#run-workflow">Run workflow</a></li>
      </ul>
</li>
      <li>
<a href="#read-preprocessing">Read preprocessing</a><ul class="nav nav-pills nav-stacked">
<li><a href="#experiment-definition-provided-by-targets-file">Experiment definition provided by <code>targets</code> file</a></li>
      <li><a href="#read-quality-filtering-and-trimming">Read quality filtering and trimming</a></li>
      <li><a href="#fastq-quality-report">FASTQ quality report</a></li>
      </ul>
</li>
      <li>
<a href="#alignments">Alignments</a><ul class="nav nav-pills nav-stacked">
<li><a href="#read-mapping-with-bowtie2">Read mapping with <code>Bowtie2</code></a></li>
      <li><a href="#read-and-alignment-stats">Read and alignment stats</a></li>
      <li><a href="#create-symbolic-links-for-viewing-bam-files-in-igv">Create symbolic links for viewing BAM files in IGV</a></li>
      </ul>
</li>
      <li>
<a href="#utilities-for-coverage-data">Utilities for coverage data</a><ul class="nav nav-pills nav-stacked">
<li><a href="#rle-object-stores-coverage-information">Rle object stores coverage information</a></li>
      <li><a href="#resizing-aligned-reads">Resizing aligned reads</a></li>
      <li><a href="#naive-peak-calling">Naive peak calling</a></li>
      <li><a href="#plot-coverage-for-defined-region">Plot coverage for defined region</a></li>
      </ul>
</li>
      <li>
<a href="#peak-calling-with-macs2">Peak calling with MACS2</a><ul class="nav nav-pills nav-stacked">
<li><a href="#merge-bam-files-of-replicates-prior-to-peak-calling">Merge BAM files of replicates prior to peak calling</a></li>
      <li><a href="#peak-calling-without-inputreference-sample">Peak calling without input/reference sample</a></li>
      <li><a href="#peak-calling-with-inputreference-sample">Peak calling with input/reference sample</a></li>
      <li><a href="#identify-consensus-peaks">Identify consensus peaks</a></li>
      </ul>
</li>
      <li>
<a href="#annotate-peaks-with-genomic-context">Annotate peaks with genomic context</a><ul class="nav nav-pills nav-stacked">
<li><a href="#annotation-with-chippeakanno-package">Annotation with <code>ChIPpeakAnno</code> package</a></li>
      <li><a href="#annotation-with-chipseeker-package">Annotation with <code>ChIPseeker</code> package</a></li>
      </ul>
</li>
      <li><a href="#count-reads-overlapping-peaks">Count reads overlapping peaks</a></li>
      <li><a href="#differential-binding-analysis">Differential binding analysis</a></li>
      <li><a href="#go-term-enrichment-analysis">GO term enrichment analysis</a></li>
      <li>
<a href="#motif-analysis">Motif analysis</a><ul class="nav nav-pills nav-stacked">
<li><a href="#parse-dna-sequences-of-peak-regions-from-genome">Parse DNA sequences of peak regions from genome</a></li>
      <li><a href="#motif-discovery-with-bcrank">Motif discovery with <code>BCRANK</code></a></li>
      </ul>
</li>
      <li><a href="#version-information">Version Information</a></li>
      <li><a href="#funding">Funding</a></li>
      <li><a href="#references">References</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://www.dcassol.com/">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '',
    indexName: '',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
